doc/kernel/other/interrupts.rst

```diff
@@ -218,6 +218,27 @@ The following code defines and enables an ISR.
        ...
     }
 
+Since the :c:macro:`IRQ_CONNECT` macro requires that all its parameters be
+known at build time, in some cases this may not be acceptable. It is also
+possible to install interrupts at runtime with
+:cpp:func:`irq_connect_dynamic()`. It is used in exactly the same way as
+:c:macro:`IRQ_CONNECT`:
+
+.. code-block:: c
+
+    void my_isr_installer(void)
+    {
+       ...
+       irq_connect_dynamic(MY_DEV_IRQ, MY_DEV_PRIO, my_isr, MY_ISR_ARG,
+                           MY_IRQ_FLAGS);
+       irq_enable(MY_DEV_IRQ);
+       ...
+    }
+
+Dynamic interrupts require the :option:`CONFIG_DYNAMIC_INTERRUPTS` option to
+be enabled. Removing or re-configuring a dynamic interrupt is currently
+unsupported.
```

Comment:
I don't see anything that prevents re-configuring by calling irq_connect_dymanic multiple times for the same irq, at least not for the generic non-x86 case.